examples/stable_diffusion_v2/cldm/cldm.py

```diff
@@ -398,3 +422,31 @@ def configure_optimizers(self):
             params += list(self.model.diffusion_model.out.get_parameters())
         opt = nn.optim.adam.AdamWeightDecay(params, learning_rate=lr)
         return opt
+
+    def p_losses(self, x_start, c_concat, c_crossattn, t, noise=None):
```

Comment:
`p_losses()` here is identical to the parent's method. I suggest you keep the unified `cond` here as well.

Reply:
The input arguments are not the same. Extra input `c_concat` is involved in this func, and I checked your PR for T2I training that adds additional input `*args` to parent's `p_losses()`. I thought it might be best for me not to change the parent's `p_losses()`.

Reply:
I think it's better to add `**kwargs` in `p_losses()` as well and don't duplicate identical code. Just overload `apply_model()` method to suit your needs.